cmd/clusterctl/client/cluster/mover.go

```diff
@@ -249,6 +338,70 @@ func (o *objectMover) move(graph *objectGraph, toProxy Proxy) error {
+	log := logf.Log
+
+	clusters := graph.getClusters()
+	log.Info("Starting backup of Cluster API objects", "Clusters", len(clusters))
+
+	// Sets the pause field on the Cluster object in the source management cluster, so the controllers stop reconciling it.
+	log.V(1).Info("Pausing the source cluster")
+	if err := setClusterPause(o.fromProxy, clusters, true, o.dryRun); err != nil {
+		return err
+	}
+
+	// Define the move sequence by processing the ownerReference chain, so we ensure that a Kubernetes object is moved only after its owners.
+	// The sequence is bases on object graph nodes, each one representing a Kubernetes object; nodes are grouped, so bulk of nodes can be moved in parallel. e.g.
+	// - All the Clusters should be moved first (group 1, processed in parallel)
+	// - All the MachineDeployments should be moved second (group 1, processed in parallel)
+	// - then all the MachineSets, then all the Machines, etc.
+	moveSequence := getMoveSequence(graph)
+
+	// Save all objects group by group
+	log.Info("Saving files to %s", directory)
+	for groupIndex := 0; groupIndex < len(moveSequence.groups); groupIndex++ {
+		if err := o.backupGroup(moveSequence.getGroup(groupIndex), directory); err != nil {
+			return err
+		}
+	}
+
+	// Reset the pause field on the Cluster object in the target management cluster, so the controllers start reconciling it.
+	log.V(1).Info("Resuming the target cluster")
+	return setClusterPause(o.fromProxy, clusters, false, o.dryRun)
+}
+
+func (o *objectMover) restore(graph *objectGraph, toProxy Proxy) error {
+	log := logf.Log
+
+	// Get clusters from graph
+	clusters := graph.getClusters()
+
+	// Ensure all the expected target namespaces are in place before creating objects.
+	log.V(1).Info("Creating target namespaces, if missing")
+	if err := o.ensureNamespaces(graph, toProxy); err != nil {
+		return err
+	}
+
+	// Define the move sequence by processing the ownerReference chain, so we ensure that a Kubernetes object is moved only after its owners.
+	// The sequence is bases on object graph nodes, each one representing a Kubernetes object; nodes are grouped, so bulk of nodes can be moved in parallel. e.g.
+	// - All the Clusters should be moved first (group 1, processed in parallel)
+	// - All the MachineDeployments should be moved second (group 1, processed in parallel)
+	// - then all the MachineSets, then all the Machines, etc.
+	moveSequence := getMoveSequence(graph)
+
+	// Create all objects group by group, ensuring all the ownerReferences are re-created.
+	log.Info("Restoring objects into the target cluster")
+	for groupIndex := 0; groupIndex < len(moveSequence.groups); groupIndex++ {
+		if err := o.restoreGroup(moveSequence.getGroup(groupIndex), toProxy); err != nil {
+			return err
+		}
+	}
+
+	// Reset the pause field on the Cluster object in the target management cluster, so the controllers start reconciling it.
+	log.V(1).Info("Resuming the source cluster")
```

Comment:
```suggestion
	log.V(1).Info("Resuming the target cluster")
```
Also, should we Pause before restoring?